src/app/server/Server.cpp

```diff
@@ -550,6 +558,15 @@ void InitServer(AppDelegate * delegate)
     err = gExchangeMgr.RegisterUnsolicitedMessageHandlerForProtocol(Protocols::ServiceProvisioning::Id, &gCallbacks);
     VerifyOrExit(err == CHIP_NO_ERROR, err = CHIP_ERROR_NO_UNSOLICITED_MESSAGE_HANDLER);
 
+#if CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONER_DISCOVERY
+    err = chip::Protocols::UserDirectedCommissioning::UserDirectedCommissioningServer::GetInstance().Init(&gExchangeMgr);
+    SuccessOrExit(err);
+#endif
+#if CHIP_DEVICE_CONFIG_ENABLE_BOTH_COMMISSIONER_AND_COMMISSIONEE
+    err = InitCommissioner();
```

Comment:
As the comment in CHIPDeviceConfig.h notes:

CHIP_DEVICE_CONFIG_ENABLE_COMMISSIONER_DISCOVERY depends upon CHIP_DEVICE_CONFIG_ENABLE_BOTH_COMMISSIONER_AND_COMMISSIONEE set to 1

This is because a device needs to be both (commissioner and Commissionee) in the code today in order for commissioner discovery to make sense. In the (near) future when we have more permutations (eg. commissioner only) then I will try to make them independent.